store/sqlstore/channel_store.go

```diff
@@ -2302,6 +2302,104 @@ func (s SqlChannelStore) performSearch(searchQuery string, term string, paramete
 	return result
 }
 
+func (s SqlChannelStore) getSearchGroupChannelsQuery(userId, term string, isPostgreSQL bool) (string, map[string]interface{}) {
+	var query, baseLikeClause string
+	if isPostgreSQL {
+		baseLikeClause = "ARRAY_TO_STRING(ARRAY_AGG(u.Username), ', ') LIKE %s"
+		query = `
+            SELECT
+                *
+            FROM
+                Channels
+            WHERE
+                Id IN (
+                    SELECT
+                        cc.Id
+                    FROM (
+                        SELECT
+                            c.Id
+                        FROM
+                            Channels c
+                        JOIN
+                            ChannelMembers cm on c.Id = cm.ChannelId
+                        JOIN
+                            Users u on u.Id = cm.UserId
+                        WHERE
+                            c.Type = 'G'
+                        AND
+                            u.Username = (SELECT Username FROM Users WHERE Id = :UserId)
```

Comment:
I think here `u.Id = :UserId` would be the same, right?

Reply:
Actually, you can remove the Users join, and use `cm.UserId = :UserId`, right?

Reply:
You're right! This comes from the first iteration of the query, where I was trying to apply both filters (user is a member and username likes) at the same time. Will fix!